### PR TITLE
[SDK app-insights] Add TelemetryInitializerMiddleware class

### PR DESCRIPTION
**Pull request by matiasroldan6**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/package-info.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/package-info.java:2]: This copyright is different to [this](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/package-info.java). This is just an example, the one of the QnA follows the rest of copyrights.


I noticed that this way of Copyright is the one of the Samples, but in the SDKs is different

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import com.microsoft.bot.builder.Middleware;
+import com.microsoft.bot.builder.NextDelegate;
+import com.microsoft.bot.builder.TelemetryLoggerMiddleware;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.schema.Activity;
+
+import java.util.concurrent.CompletableFuture;
+
+public class TelemetryInitializerMiddleware implements Middleware {
+    private HttpContext httpContext;
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java:18]:
```suggestion

    private HttpContext httpContext;
```

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private TelemetryLoggerMiddleware telemetryLoggerMiddleware;
+    private Boolean logActivityTelemetry;
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java:20]: These are `final` as in [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/integration/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/TelemetryInitializerMiddleware.cs#L18) they are `readonly`

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    public TelemetryInitializerMiddleware(TelemetryLoggerMiddleware withTelemetryLoggerMiddleware, Boolean withLogActivityTelemetry) {
+        telemetryLoggerMiddleware = withTelemetryLoggerMiddleware;
+        logActivityTelemetry = withLogActivityTelemetry != null ? withLogActivityTelemetry : true;
+    }
+
+    public CompletableFuture<Void> onTurn(TurnContext context, NextDelegate next) {
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java:27]:
```suggestion
    public CompletableFuture<Void> onTurn(TurnContext context, NextDelegate nextTurn) {
```

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                httpContext.removeAttribute(TelemetryBotIdInitializer.BotActivityKey);
+            }
+
+            httpContext.setAttribute(TelemetryBotIdInitializer.BotActivityKey, activity);
+        }
+
+        if (logActivityTelemetry) {
+            return telemetryLoggerMiddleware.onTurn(context, next);
+        } else {
+            return next.next();
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java:49]:
```suggestion
            return nextTurn.next();
```

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+            if (item != null) {
+                httpContext.removeAttribute(TelemetryBotIdInitializer.BotActivityKey);
+            }
+
+            httpContext.setAttribute(TelemetryBotIdInitializer.BotActivityKey, activity);
+        }
+
+        if (logActivityTelemetry) {
+            return telemetryLoggerMiddleware.onTurn(context, next);
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java:47]:
```suggestion
            return telemetryLoggerMiddleware.onTurn(context, nextTurn);
```

**Review by Batta32 — CHANGES_REQUESTED** [review]

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java (hunk 4 above)
Comment (matiasroldan6) [reply libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java:27]: checkout that in [other](https://github.com/southworks/botbuilder-java/blob/feature/southworks/libraries/application-insights/telemetry-initializer-middleware/libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TranscriptLoggerMiddleware.java#L57) middlewares it's called [next](https://github.com/southworks/botbuilder-java/blob/feature/southworks/libraries/application-insights/telemetry-initializer-middleware/libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TelemetryLoggerMiddleware.java#L72)

Comment (Batta32) [reply libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/core/TelemetryInitializerMiddleware.java:27]: Ack, so let's do it as `next`, no change needed here

**Review by Batta32 — APPROVED** [review]

**Pull request merged**